internal/server/handlers.go

```diff
@@ -637,5 +628,35 @@ func (a *API) providerClient(c *gin.Context, provider *models.Provider, redirect
 		return nil, fmt.Errorf("client secret not found")
 	}
 
-	return providers.NewOIDC(*provider, clientSecret, redirectURL), nil
+	return providers.NewOIDCClient(*provider, clientSecret, redirectURL), nil
+}
+
+// setProviderInfoFromServer checks information provided by an OIDC server
+func (a *API) setProviderInforFromServer(c *gin.Context, provider *models.Provider) error {
```

Comment:
The API provider creation/update doesn't specify the auth URL and scopes. We look them up here. This means the API is backwards compatible.

We can assume that the `/.well-known/openid-configuration` endpoint is publicly accessible, or clients would not be able to login.